### PR TITLE
R4 Appointment Write reasonCode enhancement

### PR DESCRIPTION
**Pull request by ahamdany**
Description
----
Adding reasonCode for R4 Appointment Creates.
 
Screenshots:
Screen Shot 2020-06-22 at 6 35 58 PM
Screen Shot 2020-06-22 at 6 35 47 PMScreen Shot 2020-06-22 at 6 34 41 PM
Screen Shot 2020-06-23 at 9 37 19 AM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: content/millennium/r4/scheduling/appointment.md [hunk 1]
```diff
@@ -316,6 +316,7 @@ _Implementation Notes_
   * `Appointment.participant` must be a list containing exactly one Patient participant and at least one Location participant.
   * `Appointment.participant.actor` must be a reference to a Patient or a Location.
   * `Appointment.participant.status` must be set to `needs-action`.
+  * `Appointment.reasonCode` must be a list containing exactly one CodeableConcept.
```
Comment (MaxPhilips) [inline content/millennium/r4/scheduling/appointment.md:319]: this applies to both proposed and booked so move it up a level

we can say 'Appointment.reasonCode, if set, must be ...'

File: lib/resources/example_json/r4_examples_appointment.rb [hunk 2]
```diff
@@ -377,6 +377,17 @@ module Resources
           },
           "status": "accepted"
         }
+      ],
+      "reasonCode": [
+        {
+          "coding": [
```
Comment (MaxPhilips) [inline lib/resources/example_json/r4_examples_appointment.rb:383]: leave coding out of examples as we consume text only

File: lib/resources/r4/appointment.yaml [hunk 3]
```diff
@@ -186,3 +186,25 @@ fields:
       ]
     }
   url: https://hl7.org/fhir/appointment-definitions.html#Appointment.requestedPeriod
+
+- name: reasonCode
```
Comment (MaxPhilips) [inline lib/resources/r4/appointment.yaml:190]: there is already an entry for reasonCode on line 67, set its action field to a list of terminology and create

File: lib/resources/r4/appointment.yaml [hunk 4]
```diff
@@ -186,3 +186,25 @@ fields:
...
+  required: 'No'
+  type: List of CodeableConcept
+  description: Coded reason this appointment is scheduled.
+  action: create
+  note: Must contain a text field.
```
Comment (MaxPhilips) [inline lib/resources/r4/appointment.yaml:195]: `reasonCode.text` is required.

File: content/millennium/r4/scheduling/appointment.md (hunk 1 above)
Comment (ahamdany) [reply content/millennium/r4/scheduling/appointment.md]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/063202ecf7b52c89b017b9b8d20412808d417975#diff-1d7df8ceb9f0ae8471323516e037c662R314)

File: lib/resources/example_json/r4_examples_appointment.rb (hunk 2 above)
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_appointment.rb]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/063202ecf7b52c89b017b9b8d20412808d417975#diff-1a83ec9f4ba450cba6a16d8bcda1890cL383)

File: lib/resources/r4/appointment.yaml (hunk 3 above)
Comment (ahamdany) [reply lib/resources/r4/appointment.yaml]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/063202ecf7b52c89b017b9b8d20412808d417975#diff-fbb215a73c336ca1ac34d91de2fb411eL70)

File: lib/resources/r4/appointment.yaml (hunk 4 above)
Comment (ahamdany) [reply lib/resources/r4/appointment.yaml]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/063202ecf7b52c89b017b9b8d20412808d417975#diff-fbb215a73c336ca1ac34d91de2fb411eR76)

File: lib/resources/example_json/r4_examples_appointment.rb (hunk 2 above)
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_appointment.rb]: take it out of all the examples including in appointment.yaml

File: lib/resources/r4/appointment.yaml (hunk 4 above)
Comment (MaxPhilips) [reply lib/resources/r4/appointment.yaml]: backtick it

File: lib/resources/example_json/r4_examples_appointment.rb (hunk 2 above)
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_appointment.rb]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/74138237b033684b95e9ebd89bbd163b23aa2d59#diff-fbb215a73c336ca1ac34d91de2fb411eL76)

File: lib/resources/r4/appointment.yaml (hunk 4 above)
Comment (ahamdany) [reply lib/resources/r4/appointment.yaml]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/74138237b033684b95e9ebd89bbd163b23aa2d59#diff-fbb215a73c336ca1ac34d91de2fb411eL76)

Comment (ahamdany) [reply lib/resources/r4/appointment.yaml]: updated again [here](https://github.com/cerner/fhir.cerner.com/commit/923e21eef058f2b4164f3ffc6340bebae8db0aca#diff-fbb215a73c336ca1ac34d91de2fb411eL76)

File: lib/resources/example_json/r4_examples_appointment.rb (hunk 2 above)
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_appointment.rb]: updated again [here](https://github.com/cerner/fhir.cerner.com/commit/74138237b033684b95e9ebd89bbd163b23aa2d59#diff-fbb215a73c336ca1ac34d91de2fb411eL81)

**Comment by MaxPhilips** [comment]
we missed a 'required' yaml key, opened https://github.com/cerner/fhir.cerner.com/issues/404

verified on the live site otherwise:

Screen Shot 2020-06-23 at 5 02 09 PM

**Pull request merged**